File: app/src/main/java/com/wielabs/ItemBackgroundDecoration.java
package com.wielabs;

import android.graphics.Color;
import android.graphics.Rect;
import android.view.View;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.mikhaellopez.circularimageview.CircularImageView;

public class ItemBackgroundDecoration extends RecyclerView.ItemDecoration {

    @Override
    public void getItemOffsets(@NonNull Rect outRect, @NonNull View view, @NonNull RecyclerView parent, @NonNull RecyclerView.State state) {
        super.getItemOffsets(outRect, view, parent, state);

        CircularImageView circularImageView = view.findViewById(R.id.profilePicLeaderBoard);

        switch (parent.getChildAdapterPosition(view)) {
            case 0:
                circularImageView.setBorderColor(Color.parseColor("#ffd94a"));
                break;

            case 1:
                circularImageView.setBorderColor(Color.parseColor("#d7d8c8"));
                break;

            case 2:
                circularImageView.setBorderColor(Color.parseColor("#f4aa6b"));
                break;

            default:
                circularImageView.setBorderColor(Color.parseColor("#f9f9f9"));
        }

//        switch (parent.getChildAdapterPosition(view)){
//            case 0 : ((GradientDrawable)view.findViewById(R.id.constraintLayout).getBackground()).setColor(Color.parseColor("#ffd94a"));
//            break;
//
//            case 1 : ((GradientDrawable)view.findViewById(R.id.constraintLayout).getBackground()).setColor(Color.parseColor("#d7d8c8"));
//            break;
//
//            case 2 : ((GradientDrawable)view.findViewById(R.id.constraintLayout).getBackground()).setColor(Color.parseColor("#f4aa6b"));
//            break;
//
//            default:((GradientDrawable)view.findViewById(R.id.constraintLayout).getBackground()).setColor(Color.parseColor("#f9f9f9"));
//        }
    }
}

File: app/src/main/java/com/wielabs/ItemBackgroundDecoration.kt
package com.example.bidbadsample

import android.content.res.ColorStateList
import android.graphics.Color
import android.graphics.Rect
import android.graphics.drawable.GradientDrawable
import android.view.View
import androidx.recyclerview.widget.RecyclerView

class ItemBackgroundDecoration : RecyclerView.ItemDecoration() {
    override fun getItemOffsets(
        outRect: Rect,
        view: View,
        parent: RecyclerView,
        state: RecyclerView.State
    ) {
        when (parent.getChildAdapterPosition(view)) {
            0 -> {
                (view.background as GradientDrawable).color =
                    ColorStateList.valueOf(Color.parseColor("#ffd94a"))
            }
            1 -> {
                (view.background as GradientDrawable).color =
                    ColorStateList.valueOf(Color.parseColor("#d7d8c8"))
            }
            2 -> {
                (view.background as GradientDrawable).color =
                    ColorStateList.valueOf(Color.parseColor("#f4aa6b"))
            }
            else -> {
                (view.background as GradientDrawable).color =
                    ColorStateList.valueOf(Color.parseColor("#f9f9f9"))
            }
        }
    }
}